src/error.rs

```diff
@@ -38,28 +43,25 @@ impl Error {
     /// Encountered an unexpected situation which should not happen in practice.
     pub const UNEXPECTED: Error = Self::new_internal(2);
 
-    /// Codes below this point represent OS Errors (i.e. positive i32 values).
-    /// Codes at or above this point, but below [`Error::CUSTOM_START`] are
-    /// reserved for use by the `rand` and `getrandom` crates.
-    pub const INTERNAL_START: u32 = 1 << 31;
-
-    /// Codes at or above this point can be used by users to define their own
-    /// custom errors.
-    pub const CUSTOM_START: u32 = (1 << 31) + (1 << 30);
+    /// Internal errors can be in the range of 2^16..2^17
+    const INTERNAL_START: RawOsError = 1 << 16;
+    /// Custom errors can be in the range of 2^17..(2^17 + 2^16)
+    const CUSTOM_START: RawOsError = 1 << 17;
 
     /// Creates a new instance of an `Error` from a particular OS error code.
     ///
     /// This method is analogous to [`std::io::Error::from_raw_os_error()`][1],
```

Comment:
This documentation now doesn't match the implementation, as we now:
  - On UEFI or `solid_asp3`: we take the raw error (which is already negative in error cases)
  - On all other platforms: we take the negative of the raw error

We should either:
  - Update the docs
  - have `Error::from_os_error` work like `Error::from_raw_os_error`

as this method isn't public, we could even change the name of `Error::from_os_error` to be something like `Error::from_neg_os_error`. I'm fine considering the UEFI error codes to be "negative" as they have the high-bit set.

Reply:
I introduced a separate method gated on UEFI targets to make it a bit more clearer.